### PR TITLE
chore: release issue template improvements

### LINKED ISSUE #3701: Improvements for release issue template
### What problem are you facing?

We have a lovely issue template that we use as a task list for each Crossplane release: 

* releases: https://github.com/crossplane/crossplane/blob/master/.github/ISSUE_TEMPLATE/release.md
* patch releases: https://github.com/crossplane/crossplane/blob/master/.github/ISSUE_TEMPLATE/patch_release.md

Going through the v1.11.0 release in #3600, we found a few improvements that we should incorporate into these templates.

* Update the reference issue in the intro section to point to the latest example of #3600 
* There is an entirely new process for docs releases, that is captured in the [docs repo release issue template](https://github.com/crossplane/docs/blob/master/.github/ISSUE_TEMPLATE/new_release.md).  We have this captured early in the crossplane release flow, but some later steps should be updated/removed
  * We no longer need an explicit "Updated the current release version in the [Crossplane docs website repo](https://github.com/crossplane/docs)."
  * We no longer need an explicit "Updated the release branch reaching EOL with https://github.com/crossplane/crossplane/pull/3003."
* We should add a task near the end to ensure that the docs release is completed, probably before announcing on slack/twitter
* Should a verification step be added to verify that the helm chart is published/promoted on the proper channels?
* We're no longer using the configurations workflow (see #3713), so this step can be removed
  * doesn't look like we have alignment/conclusion on this one, so skip removing this for now

@phisco, what other feedback did you have to add to this issue?

### PR DESCRIPTION
**Pull request by phisco**
Signed-off-by: Philippe Scorsolini <p.scorsolini@gmail.com>

### Description of your changes

Fixes #3701 

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested

N.A.

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

**Comment by jbw976** [comment]
Out of curiousity, what does `chore:` mean in the title of this PR? do we have that convention documented anywhere? 🤔

**Comment by phisco** [comment]
Yes, @jbw976, sorry I'm used to use that, it's following [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/). It's not strongly dictating all the types, so chore is not one of the ones explicitly defined in the spec, but it's a pretty common one. It's usually used to show that it's not touching actual code. Could have been also a docs, probably.

File: .github/ISSUE_TEMPLATE/release.md [hunk 1]
```diff
@@ -18,31 +18,29 @@ Please assign the release manager to the issue.
 
 This issue can be closed when we have completed the following steps (in order).
 Please ensure all artifacts (PRs, workflow runs, Tweets, etc) are linked from
-this issue for posterity. Refer to this [prior release issue][release-1.7] for
-examples of each step.
+this issue for posterity. Refer to this [prior release issue][release-1.11] for
+examples of each step, assuming release vX.Y.0 is being cut.
```
Comment (jbw976) [inline .github/ISSUE_TEMPLATE/release.md:22]: We should also update the patch release issue template as part of this PR, so both are up to date :)

https://github.com/crossplane/crossplane/blob/master/.github/ISSUE_TEMPLATE/patch_release.md

File: .github/ISSUE_TEMPLATE/release.md [hunk 2]
```diff
@@ -18,31 +18,29 @@ Please assign the release manager to the issue.
...
-- [ ] Update the `baseBranches` list in renovate's configuration file, `.github/renovate.json5`, with the new release branch, removing the old unsupported release one.
+- [ ] Run the [Promote workflow][promote-workflow] with channel `stable` on the `release-X.Y` branch and verified that the tagged build version exists on the [releases.crossplane.io] `stable` channel at `stable/vX.Y.0/...`.
+- [ ] Published a [new release] for the tagged version, with the same name as the version and descriptive release notes, taking care of generating the changes list selecting as "Previous tag" `vX.Y-1.0`, so the first of the releases for the previous minor.
+- [ ] Updated the [`crossplane/test` repo test workflows][crossplane-test-workflows] to ensure the checkout release branch and helm install version(s) point at the new release
+- [ ] Checked that the [docs release issue] created previously has been completed.
+- [ ] Updated the [releases table] in the `README.md` and the `baseBranches` list in `.github/renovate.json5` on `master`.
```
Comment (jbw976) [inline .github/ISSUE_TEMPLATE/release.md:40]: maybe add back in the extra note we used to have here? e.g. ", removing the now old and unsupported version"

Comment (jbw976) [reply .github/ISSUE_TEMPLATE/release.md:40]: oh and explicitly mention "in a single PR" so folks make sure to streamline here :)

**Review by jbw976 — COMMENTED** [review]
looks like this captures most everything in #3701 @phisco, great work!!

Let's clean up a couple things for my comments (esp. doing the patch release issue template also) then get this merged in so we can use it in our next releases 🏁

File: .github/ISSUE_TEMPLATE/release.md (hunk 2 above)
Comment (phisco) [reply .github/ISSUE_TEMPLATE/release.md]: sounds reasonable, done!

File: .github/ISSUE_TEMPLATE/release.md (hunk 1 above)
Comment (phisco) [reply .github/ISSUE_TEMPLATE/release.md:22]: I gave it a shot, please have a look at it, I also updated/cleaned up the minor patch release doc once more.

File: .github/ISSUE_TEMPLATE/release.md [hunk 3]
```diff
@@ -18,46 +18,43 @@ Please assign the release manager to the issue.
...
+[releases.crossplane.io]: https://releases.crossplane.io
 [tag-workflow]: https://github.com/crossplane/crossplane/actions/workflows/tag.yml
-[ci-workflow]: https://github.com/crossplane/crossplane/actions/workflows/ci.yml
-[configurations-workflow]: https://github.com/crossplane/crossplane/actions/workflows/configurations.yml
-[promote-workflow]: https://github.com/crossplane/crossplane/actions/workflows/promote.yml
-[crossplane-test-workflows]: https://github.com/crossplane/test/tree/master/.github/workflows
-[release-1.7]: https://github.com/crossplane/crossplane/issues/2977
-[Code Freeze]: https://crossplane.io/docs/master/reference/release-cycle.html#code-freeze
\ No newline at end of file
+[xpkg.upbound.io]: https://cloud.upbound.io/browse
```
Comment (jbw976) [inline .github/ISSUE_TEMPLATE/release.md:60]: what do you think about updating this link to https://marketplace.upbound.io/configurations?query=getting-started, both here and in the patch release template?  this could be a more direct way to find the released configurations

File: .github/ISSUE_TEMPLATE/release.md [hunk 4]
```diff
@@ -18,46 +18,43 @@ Please assign the release manager to the issue.
...
+- [ ] Run the [Promote workflow][promote-workflow] with channel `stable` on the `release-X.Y` branch and verified that the tagged build version exists on the [releases.crossplane.io] `stable` channel at `stable/vX.Y.0/...`.
+- [ ] Published a [new release] for the tagged version, with the same name as the version and descriptive release notes, taking care of generating the changes list selecting as "Previous tag" `vX.<Y-1>.0`, so the first of the releases for the previous minor.
+- [ ] Checked that the [docs release issue] created previously has been completed.
+- [ ] Updated, in a single PR, the following on `master`:
+  - [ ] The [releases table] in the `README.md`, removing the now old unsupported release and adding the new one.
+  - [ ] The `baseBranches` list in `.github/renovate.json5`, removing the now old unsupported release and adding the new one.
+  - [ ] The [`crossplane/test` repo test workflows][crossplane-test-workflows] to ensure the checkout release branch and helm install version(s) point at the new release across all the workflow files.
```
Comment (jbw976) [inline .github/ISSUE_TEMPLATE/release.md:42]: i think this task belongs outside of the parent "in a single PR", because it's in a different repo entirely and therefore must be done in a separate PR 😉

File: .github/ISSUE_TEMPLATE/release.md [hunk 5]
```diff
@@ -18,46 +18,43 @@ Please assign the release manager to the issue.
...
 - [ ] Ensured that users have been notified of the release on all communication channels:
   - [ ] Slack
   - [ ] Twitter
-- [ ] Updated the [releases table] in the `README.md` on `master`.
-- [ ] Updated the current release version in the [Crossplane docs website repo].
-- [ ] Updated the release branch reaching EOL with [docs removal directive].
-- [ ] Request @jbw976 to remove the EOL docs version from Google Search
```
Comment (jbw976) [inline .github/ISSUE_TEMPLATE/release.md:46]: Actually this step will still be needed I think - I would still go into the google search console and add a removal request for e.g. https://docs.crossplane.io/v1.9

**Review by jbw976 — CHANGES_REQUESTED** [review]
awesome @phisco thanks for the quick turnaround on the feedback and taking a stab at the patch release template too!

A few more tweaks to make and then we should be good!! 🙇

**Comment by phisco** [comment]
@jbw976, addressed your comments, you can have another look at it, let me know if there is anything else, otherwise I'd squash everything before merging it 🙏

File: .github/ISSUE_TEMPLATE/release.md [hunk 6]
```diff
@@ -18,46 +18,44 @@ Please assign the release manager to the issue.
...
+  - [ ] The `baseBranches` list in `.github/renovate.json5`, removing the now old unsupported release and adding the new one.
+- [ ] Updated the [`crossplane/test` repo test workflows][crossplane-test-workflows] to ensure the checkout release branch and helm install version(s) point at the new release across all the workflow files.
 - [ ] Ensured that users have been notified of the release on all communication channels:
   - [ ] Slack
   - [ ] Twitter
-- [ ] Updated the [releases table] in the `README.md` on `master`.
-- [ ] Updated the current release version in the [Crossplane docs website repo].
-- [ ] Updated the release branch reaching EOL with [docs removal directive].
-- [ ] Request @jbw976 to remove the EOL docs version from Google Search
+-- [ ] Request @jbw976 to remove the EOL docs version from Google Search
```
Comment (jbw976) [inline .github/ISSUE_TEMPLATE/release.md:46]: looks like the formatting is broken here, this list item isn't on its own new line - probably because of the double hyphen to start the line?

File: .github/ISSUE_TEMPLATE/patch_release.md [hunk 7]
```diff
@@ -18,31 +18,32 @@ Please assign the release manager to the issue.
 
 This issue can be closed when we have completed the following steps (in order).
 Please ensure all artifacts (PRs, workflow runs, Tweets, etc) are linked from
-this issue for posterity. Refer to this [prior release issue][release-1.7] for
-examples of each step.
+this issue for posterity. Refer to this [prior release issue][release-1.11.1] for
+examples of each step, assuming vX.Y.Z is being cut.
 
 - [ ] Updated all version information in the documentation on the relevant release branch.
```
Comment (jbw976) [inline .github/ISSUE_TEMPLATE/patch_release.md:24]: From https://github.com/crossplane/crossplane/pull/3702#discussion_r1117488438, I think this can be removed too - the docs have nothing at all that needs to be done for a patch release 😅

**Review by jbw976 — APPROVED** [review]
2 final things to address and then please do go ahead and get this squashed and merged! thanks @phisco!!

**Review by hasheddan — APPROVED** [review]
